src/audio/graph_thread.rs

```diff
@@ -1,24 +1,25 @@
+use audio::block::Chunk;
 use audio::node::{AudioNodeEngine, AudioNodeType};
+use audio::gain_node::GainNode;
 use audio::oscillator_node::OscillatorNode;
 use audio::sink::AudioSink;
 use std::cell::RefCell;
-use std::collections::hash_map::HashMap;
 use std::sync::mpsc::Receiver;
 use std::sync::Arc;
 
 #[cfg(feature = "gst")]
 use backends::gstreamer::audio_sink::GStreamerAudioSink;
 
 pub enum AudioGraphThreadMsg {
-    CreateNode(usize, AudioNodeType),
+    CreateNode(AudioNodeType),
     ResumeProcessing,
     PauseProcessing,
 }
 
 pub struct AudioGraphThread {
     // XXX Test with a hash map for now. This should end up
     // being a graph, like https://docs.rs/petgraph/0.4.12/petgraph/.
-    nodes: RefCell<HashMap<usize, Box<AudioNodeEngine>>>,
+    nodes: RefCell<Vec<Box<AudioNodeEngine>>>,
```

Comment:
Sorry, I forgot to mention this. Can we also leave the HashMap until we use something like [petgraph.GraphMap](https://docs.rs/petgraph/0.4.12/petgraph/graphmap/struct.GraphMap.html), please? We need it to implement node methods like #21.

Reply:
I meant leaving both, the HashMap and the Vec. Or maybe use something like (linked_hash_map)[https://docs.rs/linked-hash-map/0.5.1/linked_hash_map/]. But I guess Vec indices would work for now.